controllers/rack.go

```diff
@@ -1731,6 +1743,75 @@ func (r *SingleClusterReconciler) getCurrentRackList() (
 	return rackList, nil
 }
 
+func (r *SingleClusterReconciler) handleEnableSecurity(rackState *RackState, ignorablePodNames sets.Set[string]) error {
+	if !r.enablingSecurity() {
+		// No need to proceed if security is not to be enabling
+		return nil
+	}
+
+	// Get pods where security-enabled config is applied
+	securityEnabledPods, err := r.getPodsWithUpdatedConfigForRack(rackState)
+	if err != nil {
+		return err
+	}
+
+	if len(securityEnabledPods) == 0 {
+		// No security-enabled pods found
+		return nil
+	}
+
+	// Setup access control.
+	if err := r.validateAndReconcileAccessControl(securityEnabledPods, ignorablePodNames); err != nil {
+		r.Log.Error(err, "Failed to Reconcile access control")
+		r.Recorder.Eventf(
+			r.aeroCluster, corev1.EventTypeWarning, "ACLUpdateFailed",
+			"Failed to setup Access Control %s/%s", r.aeroCluster.Namespace,
+			r.aeroCluster.Name,
+		)
+
+		return err
+	}
+
+	return nil
+}
+
+func (r *SingleClusterReconciler) enablingSecurity() bool {
+	return r.aeroCluster.Spec.AerospikeAccessControl != nil && r.aeroCluster.Status.AerospikeAccessControl == nil
+}
+
+func (r *SingleClusterReconciler) getPodsWithUpdatedConfigForRack(rackState *RackState) ([]corev1.Pod, error) {
+	pods, err := r.getOrderedRackPodList(rackState.Rack.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list pods: %v", err)
+	}
+
+	if len(pods) == 0 {
+		// No pod found for the rack
+		return nil, nil
+	}
+
+	confMap, err := r.getConfigMap(rackState.Rack.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	requiredConfHash := confMap.Data[aerospikeConfHashFileName]
+
+	securityEnabledPods := make([]corev1.Pod, 0, len(r.aeroCluster.Status.Pods))
```

Comment:
```suggestion
	securityEnabledPods := make([]corev1.Pod, 0, len(pods))
```

This should be len of rack pods as status contains pods of all racks